sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -954,26 +962,66 @@ public void onPersonChanged(PersonDto existingPerson, Person newPerson) {
 
 		// Update cases if present condition has changed
 		if (existingPerson != null) {
+			// sort cases based on recency
+			Collections.sort(
+				personCases,
+				(c1, c2) -> CaseLogic.getStartDate(c1.getSymptoms().getOnsetDate(), c1.getReportDate())
+					.before(CaseLogic.getStartDate(c2.getSymptoms().getOnsetDate(), c2.getReportDate())) ? 1 : -1);
+
 			if (newPerson.getPresentCondition() != null && existingPerson.getPresentCondition() != newPerson.getPresentCondition()) {
 				// Update case list after previous onCaseChanged
-				personCases = caseService.findBy(new CaseCriteria().person(new PersonReferenceDto(newPerson.getUuid())), true);
```

Comment:
If I'm not mistaken, you still need to refresh the personCases here because onCaseChanged might have changed something about these cases, and you won't know these changes without querying for them again - or is that incorrect?